### PR TITLE
Validate current OS is tier 1 + unit tests

### PR DESCRIPTION
**Pull request by marianan**

### CODE REVIEW

File: src/azure-iot-edge-installer.sh [hunk 1]
```diff
@@ -27,8 +28,16 @@ chmod +x install-container-management.sh
 chmod +x install-edge-runtime.sh
 chmod +x validate-post-install.sh
 
+# check if current OS is Tier 1
+. /etc/os-release
+is_os_tier1 $ID $VERSION_ID
+if [ "$?" != "0" ]
+then 
+	echo "This OS is not supported. Exit."
+	# log_error "This OS is not supported. Exit."
```
Comment (mhshami01) [inline src/azure-iot-edge-installer.sh:37]: We can use the logger now :-)

File: src/azure-iot-edge-installer.sh [hunk 2]
```diff
@@ -33,8 +33,15 @@ chmod +x install-container-management.sh
 chmod +x install-edge-runtime.sh
 chmod +x validate-post-install.sh
 
+# check if current OS is Tier 1
+. /etc/os-release
+is_os_tier1 $ID $VERSION_ID
+if [ "$?" != "0" ]
+then 
+    log_error "This OS is not supported. Exit."
```
Comment (cindydeng1998) [inline src/azure-iot-edge-installer.sh:41]: nit: should we leave a link to the Tier 1 OS documentation, or show that only two OS are supported today?

**Review by cindydeng1998 — APPROVED** [review]
🌞

File: src/azure-iot-edge-installer.sh [hunk 3]
```diff
@@ -20,7 +20,8 @@ download_bash_script() {
         log_info "downloading '%s'" $file_name
 
         # attempt to download to a temporary file.
-        wget $url_text -q -O $tmp_file
+        #wget $url_text -q -O $tmp_file
+		cp ../$file_name .
```
Comment (mhshami01) [inline src/azure-iot-edge-installer.sh:24]: - I assume this is only temporary :-)
- Please merge with 'main' and use:
cp ../$TOPDIR/$file_name .
- nit: replace tabs with spaces :-)

File: src/azure-iot-edge-installer.sh [hunk 4]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 
 # import utils
-source src/utils.sh
-source src/validate-tier1-os.sh
+source utils.sh
+source validate-tier1-os.sh
```
Comment (mhshami01) [inline src/azure-iot-edge-installer.sh:5]: - after merging with main, use $TOPDIR/validate-tier1-os.sh

**Review by mhshami01 — APPROVED** [review]

**Pull request merged**